internal/fluentbit/config/builder/secrets_config_builder.go

```diff
@@ -0,0 +1,199 @@
+package builder
+
+import (
+	"bytes"
+	"context"
+	"fmt"
+	"maps"
+
+	corev1 "k8s.io/api/core/v1"
+	"k8s.io/apimachinery/pkg/types"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+
+	telemetryv1alpha1 "github.com/kyma-project/telemetry-manager/apis/telemetry/v1alpha1"
+	logpipelineutils "github.com/kyma-project/telemetry-manager/internal/utils/logpipeline"
+	sharedtypesutils "github.com/kyma-project/telemetry-manager/internal/utils/sharedtypes"
+)
+
+func (b *ConfigBuilder) BuildEnvConfigSecret(ctx context.Context, logPipelines []telemetryv1alpha1.LogPipeline) (map[string][]byte, error) {
+	envConfigSecret := make(map[string][]byte)
+
+	for i := range logPipelines {
+		if !logPipelines[i].DeletionTimestamp.IsZero() {
```

Comment:
Do we need to check for deletion timestamp here?
Isn't it already checked in the reconciler https://github.com/kyma-project/telemetry-manager/blob/main/internal/reconciler/logpipeline/fluentbit/reconciler.go#L174?